### PR TITLE
used cached session for connecting to Gregor, unless we have received a bad session error from it recently

### PR DESCRIPTION
**Pull request by mmaxim**
cc @maxtaco

### CODE REVIEW

**Comment by mmaxim** [comment]
I tried this manually locally and it worked (just deleted my session from DB and restarted authd+gregrod and watched logs)

**Comment by mmaxim** [comment]
Actually I'll add @maxtaco since @patrickxb wrote half of this.

**Comment by patrickxb** [comment]
Ok, sounds good.

File: go/service/gregor.go [hunk 1]
```diff
@@ -188,12 +189,13 @@ func (db *gregorLocalDb) Load(u gregor.UID) (res []byte, e error) {
...
-		pushStateFilter: func(m gregor.Message) bool { return true },
-		badger:          nil,
-		broadcastCh:     make(chan gregor1.Message, 10000),
+		Contextified:      globals.NewContextified(g),
+		chatLog:           utils.NewDebugLabeler(g, "PushHandler", false),
+		firstConnect:      true,
+		pushStateFilter:   func(m gregor.Message) bool { return true },
+		badger:            nil,
+		broadcastCh:       make(chan gregor1.Message, 10000),
+		forceSessionCheck: false,
```
Comment (patrickxb) [inline go/service/gregor.go]: This LGTM

**Pull request merged**